src/Caching/Impl/PythonDbModule.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Collections.Generic;
+using Microsoft.Python.Analysis.Caching.Models;
+using Microsoft.Python.Analysis.Modules;
+using Microsoft.Python.Core;
+
+namespace Microsoft.Python.Analysis.Caching {
+    internal sealed class PythonDbModule : SpecializedModule {
```

Comment:
Why we are deriving from `SpecializedModule` rather than from `PythonModule`?

Reply:
Just seemed to be a reasonable choice since specialized is already handled in analysis as something that does not have content or AST and specialized functions are not expected to have bodies or otherwise evaluated.

Reply:
Got it